### ISSUE TITLE
Should all AES ciphers being banned when "!AES" is given as cipher list?

### ISSUE THREAD

**Issue by qunying** [openssl/openssl#10936]
Using system provides OpenSSL 1.1.1d under Slackware64-current.

 ```
 $ openssl ciphers -v 'HIGH:!aNULL:!eNULL:!PSK:!SRP:!AES' | grep 
TLS_AES_256_GCM_SHA384  TLSv1.3 Kx=any      Au=any  Enc=AESGCM(256) Mac=AEAD
TLS_AES_128_GCM_SHA256  TLSv1.3 Kx=any      Au=any  Enc=AESGCM(128) Mac=AEAD
 ```
There are still two AES entries listed, all other AES cipher suite either CBC or GCM/CCM modes are removed.

Is this a bug?

**Comment by mattcaswell** [comment]
No, this is working as designed. The ciphersuites you are seeing are TLSv1.3 ciphersuites.

TLSv1.2 (and below) ciphersuites are handled separately to TLSv1.3 ciphersuites. The two are incompatible with each other (you can't use a TLSv1.2 ciphersuite in TLSv1.3 and vice versa).

The `openssl ciphers` command lists all TLSv1.2 and TLSv1.3 ciphersuites that are configured. However the cipherstring that you are passing to the command only controls the TLSv1.2 (and below) ciphersuites. In order to modify the TLSv1.3 ciphersuites you have to use the "-ciphersuites" flag. For details see this page:

https://www.openssl.org/docs/man1.1.1/man1/openssl-ciphers.html

**Comment by nhorman** [comment]
Marking as inactive, to be closed at the end of 3.4 dev barring further input
